pkg/operator/ceph/pool/controller.go

```diff
@@ -341,11 +347,31 @@ func (r *ReconcileCephBlockPool) reconcileCreatePool(clusterInfo *cephclient.Clu
 	return reconcile.Result{}, nil
 }
 
+func getCephName(pool cephv1.CephBlockPool) (string, error) {
```

Comment:
Should this be part of the API so it can be used for validation by the `ceph/client` or other places as well?

Reply:
`ceph/client` doesn't have a concept of the difference between the k8s and ceph pool names, so it seems correct to keep the conversion of k8s name to ceph pool name here in the controller.

Reply:
Right. But we could add a method to the API so that `CephBlockPool` has a `.Pool()` method that the client could use, and we could do sanitization of the pool name there to ensure any internal usage isn't breaking our own rules.